packages/@magic-sdk/provider/test/spec/modules/base-module/request.spec.ts

```diff
@@ -203,3 +203,103 @@ test('Ignores events with malformed response', (done) => {
     '*',
   );
 });
+
+test('Resolves with an intermediary event', async () => {
+  const expectedResult = 'monkey brains';
+  const expectedPayload = [
+    'MAGIC_HANDLE_REQUEST',
+    {
+      id: (requestPayload.id as number) + 1,
+      jsonrpc: requestPayload.jsonrpc,
+      method: 'magic_intermediary_event',
```

Comment:
QQ, how would the SDK know which root method the intermediary event is associated with?

Reply:
The arguments for the call include the id of the root method in it.